subprojects/diagnostics/src/main/java/org/gradle/api/tasks/diagnostics/internal/graph/nodes/AbstractRenderableModuleResult.java

```diff
@@ -49,7 +50,7 @@ public String getName() {
 
     @Override
     public List<ResolvedVariantResult> getAllVariants() {
-        return module.getAllVariants();
+        return ((ResolvedComponentResultInternal) module).getAllVariants();
```

Comment:
Are we sure this cast is safe always and doesn't need an instanceof check?

Reply:
There is only one implementation of `ResolvedComponentResult`, `DefaultResolvedComponentResult`, and it implements this interface. I don't think a check is needed.